### DISCUSSION TITLE
`useStyle` configurable head / style wrapper to allow use in shadow DOM

### DISCUSSION THREAD

**Discussion by mondaychen** [primefaces/community#630] [PrimeReact] [answered]
I'm building an app inside of a [shadow DOM](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM) so that the CSS would conflict with other parts of the page. 
Currently, `useStyle` appends styles into `document.head` https://github.com/primefaces/primereact/blob/01c6ef2c035b93845d64bb516c03267461f9f613/components/lib/hooks/useStyle.js#L30
This would not work well with shadow DOM because it requires adding its styles inside of ShadowRoot.

Can we make it configurable via `PrimeReactProvider`, similar to `appendTo`?

If this sounds good, I already have the patch and can submit a PR

**Comment by melloware** [comment, marked as answer]
I think your issue is similar to this? https://github.com/primefaces/primereact/issues/5246

If not can you open a new Issue and then link your proposed PR to that?

**Reply by mondaychen** [reply]
Ah yes. It should be the same thing.
I'll submit a PR within this week!

**Reply by mondaychen** [reply]
@melloware do you think `styleContainer` is a good name for the new value in PrimeReactProvider for this purpose?

**Reply by melloware** [reply]
Sure

**Reply by mondaychen** [reply]
PR: https://github.com/primefaces/primereact/pull/5566

**Reply by melloware** [reply]
Looks good I will test it out tomorrow!
